### PR TITLE
CAP-26 Created User login API using Passport.js

### PR DESCRIPTION
**Pull request by snaanortey**
Please review this PR. Here is a summary of what I did:

1. Created the `findByEmail` method in the `userService` file to find a user's data in the DB using the user's email
2. Created an API for user login in the `login` file under the routes directory and used `passport.js` to create and verify a `JWT token` for the user upon login
3. Tested that the token works by creating a secure route in the `secret.ts` file under the `routes` directory

### CODE REVIEW

File: server/src/routes/login.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { Strategy as LocalStrategy, VerifyFunction } from "passport-local";
+import bcrypt from "bcrypt";
+import { UserService } from "../services/userService";
+import { EntityNotFoundError } from "typeorm";
+import jwt from "jsonwebtoken";
+import { Strategy, ExtractJwt } from "passport-jwt";
+
+const router = express.Router();
+
+const userService = new UserService();
```
Comment (ehimah) [inline server/src/routes/login.ts:14]: please sort your imports

In the  file in VSCode, Right-click -> Source actions -> Organize imports

**Review by ehimah — CHANGES_REQUESTED** [review]

File: server/package.json [hunk 2]
```diff
@@ -45,7 +52,12 @@
     "@types/cors": "^2.8.12",
     "@types/express": "^4.17.14",
     "@types/express-fileupload": "^1.4.1",
+    "@types/express-session": "^1.17.5",
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/package.json]: addressed comment. Thanks!

File: server/package.json [hunk 3]
```diff
@@ -32,7 +33,13 @@
     "express": "^4.18.2",
     "express-async-errors": "^3.1.1",
     "express-fileupload": "^1.4.0",
+    "express-session": "^1.17.3",
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/package.json]: addressed comment. Thanks!

File: server/src/routes/login.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { config } from "dotenv";
+config();
```
[the remark this answers is not in the document]
Comment (snaanortey) [reply server/src/routes/login.ts]: addressed comment. Thanks!

File: server/src/routes/login.ts (hunk 1 above)
Comment (snaanortey) [reply server/src/routes/login.ts]: addressed

**Comment by snaanortey** [comment]
I have addressed all comments. Thanks!

File: server/.env.example [hunk 5]
```diff
@@ -7,4 +7,7 @@ DATABASE_HOST="localhost"
 DATABASE_PORT=5432
 DATABASE_USERNAME="postgres"
 DATABASE_PASSWORD=""
-DATABASE_NAME="foodease" 
\ No newline at end of file
+DATABASE_NAME="foodease" 
+
+# Login secret key signed with JWT token during user login
+JWT_KEY="ChooseYourOwnSecretKey"
```
Comment (ehimah) [inline server/.env.example:13]:
```suggestion
#  Secret key used for signing and decrypting JWT token
JWT_KEY="ChooseYourOwnSecretKey"
```

File: server/src/app.ts [hunk 6]
```diff
@@ -1,14 +1,17 @@
-import express from "express";
+import bodyParser from "body-parser";
 import cors from "cors";
-import fileUpload from "express-fileupload";
+import express from "express";
 import "express-async-errors";
-import bodyParser from "body-parser";
+import fileUpload from "express-fileupload";
+import passport from "passport";
```
Comment (ehimah) [inline server/src/app.ts:6]: 👌

File: server/src/routes/login.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import { ExtractJwt, Strategy } from "passport-jwt";
+import { Strategy as LocalStrategy, VerifyFunction } from "passport-local";
+import { EntityNotFoundError } from "typeorm";
+import { UserService } from "../services/userService";
+
+const router = express.Router();
+
+const userService = new UserService();
+
+// This link proviides more details on workflow: https://www.digitalocean.com/community/tutorials/api-authentication-with-json-web-tokensjwt-and-passport
```
Comment (ehimah) [inline server/src/routes/login.ts:14]:
```suggestion
// This link provides more details on workflow: https://www.digitalocean.com/community/tutorials/api-authentication-with-json-web-tokensjwt-and-passport
```

**Review by ehimah — APPROVED** [review]
LGTM.

update the comments and ✅

**Pull request merged**